src/ServerlessAutoSwagger.ts

```diff
@@ -1,398 +1,501 @@
-
-    generatePaths = () => {
-        const functions = this.serverless.service.functions;
-        Object.entries(functions).map(([functionName, config]) => {
-            const events = config.events || [];
-            events
-                .filter(event => {
-                    if (!((event as HttpEvent).http || (event as HttpApiEvent).httpApi)) {
-                        return false;
-                    }
-
-                    const http = (event as HttpEvent).http || (event as HttpApiEvent).httpApi;
-
-                    if (typeof http === 'string') {
-                        return false;
-                    }
-
-                    return !http.exclude;
-                })
-                .map(event => {
-                    let http = (event as HttpEvent).http || (event as HttpApiEvent).httpApi;
-                    if (typeof http === 'string') {
-                        // TODO they're using the shorthand - parse that into object.
-                        return;
-                    }
-
-                    let path = http.path;
-                    if (path[0] !== '/') path = `/${path}`;
-
-                    if (!this.swagger.paths[path]) {
-                        this.swagger.paths[path] = {};
-                    }
-
-                    this.swagger.paths[path][http.method] = {
-                        summary: functionName,
-                        description: http.description || '',
-                        tags: http.swaggerTags,
-                        operationId: functionName,
-                        consumes: ['application/json'],
-                        produces: ['application/json'],
-                        parameters: this.httpEventToParameters(http),
-                        responses: this.formatResponses(http.responseData || http.responses),
-                        security: this.httpEventToSecurity(http),
-                    };
-                });
-        });
-    };
-
-    formatResponses = (responseData: HttpResponses | undefined) => {
-        if (!responseData) {
-            // could throw error
-            return {
-                200: {
-                    description: '200 response',
-                },
-            };
-        }
-        const formatted: { [key: string]: Response } = {};
-        Object.entries(responseData).map(([statusCode, responseDetails]) => {
-            if (typeof responseDetails == 'string') {
-                formatted[statusCode] = {
-                    description: responseDetails,
-                };
-                return;
-            }
-            let response: Response = {
-                description: responseDetails.description || `${statusCode} response`,
-            };
-            if (responseDetails.bodyType) {
-                response.schema = { $ref: `#/definitions/${responseDetails.bodyType}` };
-            }
+    if (apiKeyName) {
+      this.swagger = {
+        ...this.swagger,
+        securityDefinitions: securityDefinitions,
```

Comment:
Another fun Typescript trick :)
```suggestion
        securityDefinitions,
```